subsys/net/ip/net_context.c

```diff
@@ -243,11 +241,10 @@ int net_context_get(sa_family_t family,
 				break;
 			}
 		}
-#endif
 
-#if defined(CONFIG_NET_CONTEXT_SYNC_RECV)
```

Comment:
We should try to minimize storage in net_context struct, so here having #ifdef makes sense.

Reply:
Except that other code now previously removed by #ifdefs in the preprocessor will point towards in6 and in members in this struct. So we need to define the in6 member so that the compiler is happy and no extra memory is used.

Reply:
> Isn't there compile error now as in net_context struct the recv_data_wait semaphore is surrounded by #ifdef's, and this PR does not touch net_context.h?

At least I didn't see one with 'sanitycheck -t net', but even if it survives the compiler and test case(s) there still might be an issue lurking.